code/starfield/starfield.cpp

```diff
@@ -563,34 +569,104 @@ void parse_startbl(const char *filename)
 
 			CHECK_END();
 
+			optional_string("#Motion Debris");
+
 			// normal debris pieces
-			while (optional_string("$Debris:")) {
+			// leaving this for retail - Mjn
+			if (check_for_string("$Debris:")) {
+
+				mprintf(("Using deprecated motion debris parsing for Default motion debris!\n"));
+				motion_debris this_debris;
+				this_debris.name = "Default";
+
+				int count = 0;
+
+				while (optional_string("$Debris:")) {
 				in_check = true;
 
-				stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
 
-				if (Num_debris_normal < MAX_DEBRIS_VCLIPS) {
-					strcpy_s(Debris_vclips_normal[Num_debris_normal++].name, name);
+					if (count < MAX_MOTION_DEBRIS_BITMAPS) {
+						strcpy_s(this_debris.bitmaps[count++].name, name);
+					} else {
+						Warning(LOCATION, "Could not load normal motion debris '%s'; maximum of %d exceeded.", name, MAX_MOTION_DEBRIS_BITMAPS);
+					}
 				}
-				else {
-					Warning(LOCATION, "Could not load normal motion debris '%s'; maximum of %d exceeded.", name, MAX_DEBRIS_VCLIPS);
+				if (count == MAX_MOTION_DEBRIS_BITMAPS) {
+					Motion_debris.push_back(this_debris);
+				} else {
+					error_display(0, "Not enough bitmaps defined for motion debris '%s'. Skipping!\n", this_debris.name.c_str());
 				}
 			}
 
 			CHECK_END();
```

Comment:
These check ends will require #end right? Aren't the new style motion debris definitions supposed to go up here too? Unless I'm mistaken, if I only wanted to use the new style definitions I would still need some extra random `#end`s before I could say them.

Reply:
I did test that and the extra #ends are not required. I'm not exactly sure I follow how those CHECK_END() methods work, but they don't seem to get in the way here.